SynCommons.pas

```diff
@@ -21321,6 +21321,8 @@ function VariantToBoolean(const V: Variant; var Value: Boolean): boolean;
     Value := TVarData(V).VBoolean;
   varInteger: // coming e.g. from GetJsonField()
     Value := TVarData(V).VInteger=1;
+  varString, varUString: // handles "true"/"false" values in JSON
+    Value := SameTextU(VariantToUTF8(V), BOOL_UTF8[True]);

```

Comment:
this will create a temporary string on the stack, which slows down this function a lot (hidden try...finally block)